api/auth.go

```diff
@@ -461,6 +461,78 @@ func teamList(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	return json.NewEncoder(w).Encode(result)
 }
 
+// title: team info
+// path: /teams/{name}
+// method: GET
+// produce: application/json
+// responses:
+//   200: Info team
+//   404: Not found
+//   401: Unauthorized
+func teamInfo(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	teamName := r.URL.Query().Get(":name")
+	team, err := auth.GetTeam(teamName)
+	if err != nil {
+		return &errors.HTTP{Code: http.StatusNotFound, Message: err.Error()}
+	}
+	canRead := permission.Check(t, permission.PermTeamRead)
+	if !canRead {
+		return permission.ErrUnauthorized
+	}
+	apps, err := app.List(&app.Filter{
+		Extra:     map[string][]string{"teams": {team.Name}},
+		TeamOwner: team.Name})
+	if err != nil {
+		return err
+	}
+	pools, err := pool.ListPoolsForTeam(team.Name)
+	if err != nil {
+		return err
+	}
+	users, err := auth.ListUsers()
+	if err != nil {
+		return err
+	}
+	cachedRoles := make(map[string]permission.Role)
+	includedUsers := make([]*apiUser, 0)
+	for _, user := range users {
+		for _, roleInstance := range user.Roles {
+			role, ok := cachedRoles[roleInstance.Name]
+			if !ok {
+				role, err := permission.FindRole(roleInstance.Name)
+				if err != nil {
+					break
+				}
+				cachedRoles[roleInstance.Name] = role
+			}
+			if role.ContextType == permission.CtxGlobal || (role.ContextType == permission.CtxTeam && roleInstance.ContextValue == team.Name) {
+				canInclude := permission.Check(t, permission.PermTeam)
+				if canInclude {
+					roleMap := make(map[string]*permission.Role)
+					perms, err := t.Permissions()
+					if err != nil {
+						break
```

Comment:
Here you can return the error so the user can handle them.

Reply:
Like this?
```golang
...
for _, roleInstance := range user.Roles {
  role, ok := cachedRoles[roleInstance.Name]
  if !ok {
   role, err := permission.FindRole(roleInstance.Name)
   if err != nil {
     return &errors.HTTP{Code: http.StatusInternalServerError, Message: err.Error()}
   }
   cachedRoles[roleInstance.Name] = role
  }
...
```

Reply:
yes, this way